src/Bicep.Core/Diagnostics/DiagnosticBuilder.cs

```diff
@@ -684,6 +684,12 @@ public ErrorDiagnostic ArgumentCountMismatch(int argumentCount, int mininumArgum
                 TextSpan,
                 "BCP118",
                 "Expected the \"{\" character or the \"if\" keyword at this location.");
+
+            public Diagnostic RuntimePropertyNotAllowed(string property, IEnumerable<string> usableProperties) => new ErrorDiagnostic(
+                TextSpan,
+                "BCP119",
+                $"The property \"{property}\" cannot be set using runtime properties. You can only reference the following properties: {ToQuotedString(usableProperties.OrderBy(s => s))}."

```

Comment:
The concept of 'run-time' vs 'deploy-time' may not necessarily mean anything to customers - I'd suggest phrasing this message as something like: `The property "<prop>" must be evaluable at the start of the deployment, and cannot depend on any values that have not yet been calculated. This is limited to <props> for resources, and <props> for modules.` Thoughts?

Reply:
Seeing how I already implemented dynamic `<prop>` populating (I don't know how we could list the props without hardcoding at some point e.g. list module `<props>` with 0 modules declared in the file), I think a good compromise would be 

```The property <prop> must be evaluable at the start of the deployment, and cannot depend on any values that have not yet been calculated. Accessible properties of <accessedSymbol> are <props>.```

This way, brevity would be on our side as well.

Reply:
What happens if there are multiple accessed symbols? e.g.
```
name: concat(mod1.outputs.foo, mod2.outputs.foo, res1.properties.foo, res2.properties.foo)
```

Or even just referencing a variable which potentially has a long chain of references involved?

Reply:
Each violation is separately highlighted now.